src/test/rgw/test_rgw_string.cc

```diff
@@ -12,6 +12,7 @@
  *
  */
 
+#include <errno.h>
```

Comment:
@cbodley 
What I meant is that both options are used in the code.
Include files that require other include files to be included first.
And include files that include those files themselves.

I guess it is up to the author how he wants to solve it.
Personally I prefer include files to include their own requirements.